src/pages/ImportWorkflow.js

```diff
@@ -45,15 +48,22 @@ const DockstoreImporter = ajaxCaller(class DockstoreImporter extends Component {
     try {
       const { path, version, ajax: { Dockstore } } = this.props
       const { descriptor } = await Dockstore.getWdl(path, version)
-      this.setState({ wdl: descriptor })
+      this.setState({ wdl: descriptor, workflowName: _.last(path.split('/')) })
     } catch (error) {
       reportError('Error loading WDL', error)
     }
   }
 
   render() {
     const { path, version } = this.props
-    const { isImporting, wdl } = this.state
+    const { isImporting, wdl, workflowName } = this.state
+
+    const workflowNameValidationDetails = () => {
+      return (validate({ workflowName }, {
+        workflowName: workflowNameValidation()
+      }))
+    }
```

Comment:
nit: Since validation is inherently 'pure', we could remove the function wrapper and just directly call `validate` here, and then use the resulting value in both places below.